### PR TITLE
Annotated All EventJS subclasses and classes.

### PR DESCRIPTION
**Pull request by Hunter19823**
-Added KubeJSEvent annotation to docs.
           -Attempted to annotated every EventJS class with appropriate event id.
           -Added GameStagesIntegration.GAMESTAGE_ADDED and GameStagesIntegration.GAMESTAGE_REMOVED
           -Added todo to remove KubeJSEvents.COMMAND_REGISTRY , KubeJSEvents.BLOCK_DROPS .
           -Added KubeJSEvents.SERVER_DATAPACK_FIRST_PRIORITY and KubeJSEvents.SERVER_DATAPACK_LAST_PRIORITY
           -Added KubeJSEvents.BLOCK_LOOT_TABLES
           -Added Deprecated annotation to BlockDropsEventJS
           -Added Deprecated annotation to CommandRegistryEventJS
           -Added Deprecated annotation OldItemTooltipEventJS

### CODE REVIEW

File: common/src/main/java/dev/latvian/kubejs/script/data/DataPackEventJS.java [hunk 1]
```diff
@@ -10,6 +12,14 @@
 /**
  * @author LatvianModder
  */
+@KubeJSEvent(
+		server = {
+				KubeJSEvents.SERVER_DATAPACK_LOW_PRIORITY,
+				KubeJSEvents.SERVER_DATAPACK_HIGH_PRIORITY,
+				"server.datapack.first",
```
Comment (MaxNeedsSnacks) [inline common/src/main/java/dev/latvian/kubejs/script/data/DataPackEventJS.java:19]: Shouldn't these be changed to the event IDs you've added to KubeJSEvents?

File: common/src/main/java/dev/latvian/kubejs/server/TagEventJS.java [hunk 2]
```diff
@@ -26,6 +27,10 @@
 /**
  * @author LatvianModder
  */
+@KubeJSEvent(
+		server = { "<ItemTypeName>.tags" },
+		client = { "<ItemTypeName>.tags" }
```
Comment (LatvianModder) [inline common/src/main/java/dev/latvian/kubejs/server/TagEventJS.java:38]: Tag event doesnt get fired on client. Also I would rather list the 4 basic tags - item, block, entity_type, fluid

File: common/src/main/java/dev/latvian/kubejs/server/TagEventJS.java [hunk 3]
```diff
@@ -26,6 +27,10 @@
 /**
  * @author LatvianModder
  */
+@KubeJSEvent(
+		server = { "<ItemTypeName>.tags" },
```
Comment (MaxNeedsSnacks) [inline common/src/main/java/dev/latvian/kubejs/server/TagEventJS.java:37]: Not 100% on this, `ItemTypeName` isn't really too expressive

File: common/src/main/java/dev/latvian/kubejs/item/OldItemTooltipEventJS.java [hunk 4]
```diff
@@ -10,6 +11,10 @@
 /**
  * @author LatvianModder
  */
+@Deprecated
+@KubeJSEvent(
+		client = { "client.item_tooltip" }
```
Comment (LatvianModder) [inline common/src/main/java/dev/latvian/kubejs/item/OldItemTooltipEventJS.java:16]: I wouldn't include deprecated events

File: common/src/main/java/dev/latvian/kubejs/script/data/DataPackEventJS.java (hunk 1 above)
Comment (LatvianModder) [inline common/src/main/java/dev/latvian/kubejs/script/data/DataPackEventJS.java:19]: here too

**Review by LatvianModder — CHANGES_REQUESTED** [review]

**Comment by Hunter19823** [comment]
You know what, I accidentally reverted the wrong changes when I was fixing my local build.
I did have all those as events earlier, let me fix that really quickly.

File: common/src/main/java/dev/latvian/kubejs/script/data/DataPackEventJS.java (hunk 1 above)
Comment (Hunter19823) [reply common/src/main/java/dev/latvian/kubejs/script/data/DataPackEventJS.java]: Fixed in the latest commit.

File: common/src/main/java/dev/latvian/kubejs/server/TagEventJS.java (hunk 3 above)
Comment (Hunter19823) [reply common/src/main/java/dev/latvian/kubejs/server/TagEventJS.java]: Fixed in latest commit.

File: common/src/main/java/dev/latvian/kubejs/item/OldItemTooltipEventJS.java (hunk 4 above)
Comment (Hunter19823) [reply common/src/main/java/dev/latvian/kubejs/item/OldItemTooltipEventJS.java]: I removed the reference to the ID, however, that event is still being fired under that ID.
Unlike the rest of the deprecated events.

**Comment by Hunter19823** [comment]
The last commit should fix everything you requested to change.
I do want to note that some events that called event.post(ID) instead of event.post(ScriptType,ID) were tricky to figure out whether they were on server/client as they often just used the world type.
For those, I just marked them as both client and server.

So you might want to take a quick look through any client-specific or server-specific events you know of to see if I accidentally marked them as both client and server inaccurately.

**Comment by Hunter19823** [comment]
I just removed the KubeJSEvents.BLOCK_DROPS since that is no longer an event.
As well as I removed my old todo comments.

**Pull request closed without merging**